README.md

```diff
@@ -33,132 +33,177 @@ Or install it yourself as:
-implementing the methods below, as needed.
-
-Here's an example of a simple extractor that logs out all GraphQL query details.
+### Define your own analyzer subclass
 
 ```ruby
-class LoggingExtractor < GraphQLMetrics::Instrumentation
-  def query_extracted(metrics, _metadata)
-    Rails.logger.debug({
-      query_string: metrics[:query_string],     # "query Project { project(name: "GraphQL") { tagline } }"
-      operation_type: metrics[:operation_type], # "query"
-      operation_name: metrics[:operation_name], # "Project"
-      duration: metrics[:duration]              # 0.1
-    })
+  class CaptureAllMetricsAnalyzer < GraphQL::Metrics::Analyzer
+    ANALYZER_NAMESPACE = :capture_all_metrics_analyzer_namespace
+
+    def initialize(query_or_multiplex)
+      super
+
+      # `query` is defined on instances of objects inheriting from GraphQL::Metrics::Analyzer
+      ns = query.context.namespace(ANALYZER_NAMESPACE)
+      ns[:simple_extractor_results] = {}
+    end
+
+    # @param metrics [Hash] Query metrics, including a few details about the query document itself, as well as runtime
+    # timings metrics, intended to be compatible with the Apollo Tracing spec:
+    # https://github.com/apollographql/apollo-tracing#response-format
+    #
+    # {
+    #   operation_type: "query",
+    #   operation_name: "PostDetails",
+    #   query_start_time: 1573833076.027327,
+    #   query_duration: 2.0207119999686256,
+    #   parsing_start_time_offset: 0.0010339999571442604,
+    #   parsing_duration: 0.0008190000080503523,
+    #   validation_start_time_offset: 0.0030819999519735575,
+    #   validation_duration: 0.01704599999357015,
+    # }
+    #
+    # You can use these metrics to track high-level query performance, along with any other details you wish to
+    # manually capture from `query` and/or `query.context`.
+    def query_extracted(metrics)
+      custom_metrics_from_context = {
+        request_id: query.context[:request_id],
+        # ...
+      }
+
+      # You can make use of captured metrics here (logging to Kafka, request logging etc.)
+      # log_metrics(:fields, metrics)
+      #
+      # Or store them on the query context:
+      store_metrics(:queries, metrics.merge(custom_metrics_from_context))
+    end
+
+    # For use after controller:
+    # class GraphQLController < ActionController::Base
+    #   def graphql_query
+    #     query_result = graphql_query.result.to_h
+    #     do_something_with_metrics(query.context[:simple_extractor_results])
+    #     render json: graphql_query.result
+    #   end
+    # end
+
+    # @param metrics [Hash] Field selection metrics, including resolver timings metrics, also adhering to the Apollo
+    # Tracing spec referred to above.
+    #
+    # `resolver_timings` is populated any time a field is resolved (which may be many times, if the field is nested
+    # within a list field e.g. a Relay connection field). There should always be at least one of these per field in
```

Comment:
Why is there always at least one? `{ foo { id } }` but `foo` is null then `id` will be extracted with no resolutions.

Reply:
Good catch. That's 100% true. Added a test to confirm.